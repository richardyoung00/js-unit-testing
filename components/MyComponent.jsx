import React, {Component} from 'react';

const pageStyle = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  marginTop: '10%',
  fontFamily: 'Roboto, sans-serif',
};

const marginStyle = {
  marginTop: '1em',
};

class MyComponent extends Component {

  constructor() {
    super();
    this.state = {
      name: '',
      output: '',
    };
  }

  setOutput() {
    this.setState({output: `Hello ${this.state.name}`})
  };

  render() {
    return (
      <div style={pageStyle}>
        <label>Enter your name:</label>
        <input onChange={(e) => this.setState({name: e.target.value})}/>
        <button style={marginStyle}
                onClick={() => this.setOutput()}>
          Say hello
        </button>
        <div style={marginStyle}>{this.state.output}</div>
      </div>
    )
  }
}

export default MyComponent;