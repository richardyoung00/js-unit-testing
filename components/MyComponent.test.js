import React from 'react';
import {mount} from 'enzyme';
import MyComponent from './MyComponent';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('renders correctly', () => {
  const component = mount(<MyComponent />).html();
  expect(component).toMatchSnapshot();
});










test('enter name and says hello', () => {
  const component = mount(<MyComponent />);

  component.find('input')
    .simulate('change', {
      target: {value: 'Richard'}
    });

  component.find('button').simulate('click');

  expect(component.text()).toContain('Hello Richard');
  expect(component.state('name')).toBe('Richard');
});