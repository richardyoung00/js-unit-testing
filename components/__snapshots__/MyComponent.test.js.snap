// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div style=\\"display: flex; margin-top: 10%; font-family: Roboto, sans-serif;\\"><label>Enter your name:</label><input><button style=\\"margin-top: 1em;\\">Say hello</button><div style=\\"margin-top: 1em;\\"></div></div>"`;
